daml/Marketplace/Distribution/Bid/Service.daml

```diff
@@ -0,0 +1,125 @@
+module Marketplace.Distribution.Bid.Service where
+
+import DA.List (head, last)
+import DA.Finance.Asset (AssetDeposit, AssetDeposit_Split(..))
+import DA.Finance.Types (Account, Asset, Id)
+import Marketplace.Settlement qualified as Settlement
+import Marketplace.Settlement (SettlementInstruction(..))
+import Marketplace.Distribution.Bid.Model qualified as Bid
+import Marketplace.Distribution.Bid.Model (Bid(..))
+import Marketplace.Trading.AllocationAccountRule qualified as Allocation
+import Marketplace.Trading.AllocationAccountRule (AllocationAccountRule(..))
+
+type S = Service
+
+template Service
+  with
+    operator : Party
+    provider : Party
+    issuer : Party
+    customer : Party
+    asset : Asset
+  where
+    signatory operator, provider, issuer, customer
+
+    key (assetId, customer) : (Id, Party)
+    maintainer key._2
+
+    let assetId = asset.id
+
+    controller customer can
+      nonconsuming SubmitBid : ContractId Bid.T
+        with
+          details : Bid.Details
+          tradingAccount : Account
+          allocationAccount : Account
+          depositCid : ContractId AssetDeposit
+        do
+          -- TODO: Check for parallism of this choice
+          -- TODO: Check bid details match whats contained in the asset deposit
+          depositCid <- exerciseByKey @AllocationAccountRule allocationAccount.id Allocation.Deposit with ..
+          create Bid with status = Bid.Pending; ..
+
+      -- TBC : Remove for now to block bidders from withdrawing bids
+      -- nonconsuming WithdrawBid : ContractId AssetDeposit
+      --   with
+      --     bidCid : ContractId Bid
+      --   do
+      --     Bid{..} <- fetchAndArchive bidCid
+      --     exerciseByKey @AllocationAccountRule allocationAccount.id Allocation.Withdraw with transferTo = tradingAccount; ..
+
+    controller provider, issuer can
+      nonconsuming UpdateStatus : ContractId Bid
+        with
+          newStatus : Bid.Status
+        do
+          (bidCid, bid) <- fetchByKey @Bid (asset.id, customer)
+          archive bidCid *> create bid with status = newStatus
```

Comment:
You're going a bit overboard with the monad combinators. In this case I think it's just clearer to write 
```
archive bidCid 
create bid with status = newStatus
```
on separate lines

Reply:
done